src/object/cli_shard.c

```diff
@@ -1891,7 +1891,7 @@ obj_shard_query_key_cb(tse_task_t *task, void *data)
 
 	if (rc != 0) {
 		if (rc == -DER_NONEXIST) {
-			D_RWLOCK_WRLOCK(&cb_args->obj->cob_lock);
+			D_SPIN_LOCK(&cb_args->obj->cob_spin);
```

Comment:
just confirm any special reason to change from cob_lock to cob_spin?

Reply:
no, but spin lock is faster in this case, isn't it?

Reply:
ok, seems cob_lock was added before to protect layout to avoid dead lock with cob_spin for other case, if no dead lock here then is fine to me.

Reply:
well the problem is that we still need to record the result (max epoch and max dkey/akey/recx) somewhere in that case for the single thread to process them. so might need to have a lock anyway to manage like a data structure to store the result. or we can have some lockless queue or something to add the result there. anyway, we can discuss this more in a different pr. thx